### DISCUSSION TITLE
Map Cuda image to Raylib Texture

### DISCUSSION THREAD

**Discussion by acanak** [raysan5/raylib#1714] [Q&A]
Is it  possible to map cuda image to Raylib Texture to avoid zero copy ?

**Comment by raysan5** [comment]
I don't know how CUDA manages image data but if it is stored in RAM, then you could probably use raylib `Image` struct. Texture data is stored on GPU VRAM.

**Reply by acanak** [reply]
I need to Generate one or more GL buffers to be shared with CUDA using;
GLuint bufferID;
// Generate a buffer ID
glGenBuffers(1,&bufferID);
// Make this the current UNPACK buffer (OpenGL is state-based)
glBindBuffer(GL_PIXEL_UNPACK_BUFFER, bufferID);
// Allocate data for the buffer
glBufferData(GL_PIXEL_UNPACK_BUFFER, width * height * 4,
NULL, GL_DYNAMIC_COPY);
then  register and un register the buffer for avoid memcopy
cudaGLRegisterBufferObject( GLuint bufObj );
cudaGLUnregisterBufferObject( GLuint bufObj );

then i can  draw the texture using shared buffer
// Allocate the texture memory. The last parameter is NULL since we only
// want to allocate memory, not initialize it
glTexImage2D( GL_TEXTURE_2D, 0, GL_RGBA8, Width, Height, 0, GL_BGRA,
GL_UNSIGNED_BYTE, NULL);
how ever, when ever i tired to use  opengl api, i am getting error on compiling steps for re-declarations.

**Reply by raysan5** [reply]
raylib Texture2D is allocated in a similar way but initialized with some image data. You probably can use it with CUDA, the OpenGL id is available at texture.id.
